cmd/kustomizepatch/runner.go

```diff
@@ -0,0 +1,204 @@
+package kustomizepatch
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	applicationv1alpha1 "github.com/giantswarm/apiextensions/v3/pkg/apis/application/v1alpha1"
+	"github.com/giantswarm/app/v4/pkg/app"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	vaultapi "github.com/hashicorp/vault/api"
+	"github.com/spf13/cobra"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/kustomize/kyaml/fn/framework"
+	"sigs.k8s.io/kustomize/kyaml/fn/framework/command"
+	"sigs.k8s.io/kustomize/kyaml/kio"
+	kyaml "sigs.k8s.io/kustomize/kyaml/yaml"
+	"sigs.k8s.io/yaml"
+
+	"github.com/giantswarm/konfigure/internal/generator"
+	"github.com/giantswarm/konfigure/internal/meta"
+)
+
+const (
+	nameSuffix          = "konfigure"
+	giantswarmNamespace = "giantswarm"
+	installationEnvVar  = "KONFIGURE_INSTALLATION"
+	sourceServiceEnvVar = "SOURCE_SERVICE"
+)
+
+type runner struct {
+	logger micrologger.Logger
+	stdout io.Writer
+	stderr io.Writer
+
+	config *config
+}
+
+func (r *runner) Run(_ *cobra.Command, _ []string) error {
+	r.config = new(config)
+
+	processor := framework.SimpleProcessor{
+		Config: r.config,
+		Filter: kio.FilterFunc(r.run),
+	}
+	pluginCmd := command.Build(processor, command.StandaloneDisabled, false)
+
+	err := pluginCmd.Execute()
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	return nil
+}
+
+func (r *runner) run(items []*kyaml.RNode) ([]*kyaml.RNode, error) {
```

Comment:
The whole `run()` command is pretty much identical to the one in `cmd/generate/runner.go`. 
Method arguments and output have been changed to match what kustomize expects, otherwise it's just config generation.